### PR TITLE
postinst: cleanup cloud-id-shim on xenial->bionic

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Test Steps

Run the new manual test script

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by orndorffgrant** [comment]
@renanrodrigo we don't have the nicer ubuntu upgrade integration test in `release-27` that would allow us to test this path.

We could write the manual script to use do-release-upgrade with a temporary ppa as an argument if we think that is worth it

**Comment by renanrodrigo** [comment]
I see - feel free to evaluate that. It is good for me as is (:

**Comment by orndorffgrant** [comment]
The failed tests are test bugs that have been fixed in `main` (because configurations on the backend have changed). We'll need to use those test fixes when we verify the release, but they don't require any application code changes/updates for 27.9.

File: sru/release-27.9/cleanup_cloud_id_shim.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
+x_deb=$1
+b_deb=$2
+series=bionic
```
Comment (lucasmoura) [inline sru/release-27.9/cleanup_cloud_id_shim.sh:3]: I think this should be xenial right ?

**Review by lucasmoura — APPROVED** [review]

**Comment by orndorffgrant** [comment]
FYI I am writing a manual test to do the full upgrade

**Comment by orndorffgrant** [comment]
Added the full upgrade test. Try it with:
```
bash sru/release-27.9/cleanup_cloud_id_shim_ppa.sh ppa:orndorffgrant/test-cleanup-shim
```

**Review by lucasmoura — APPROVED** [review]
New test script also worked here

File: debian/ubuntu-advantage-tools.postinst [hunk 2]
```diff
@@ -354,6 +356,17 @@ remove_old_systemd_timers() {
         deb-systemd-helper purge ua-license-check.path > /dev/null || true
         deb-systemd-helper unmask ua-license-check.path > /dev/null || true
     fi
+
+    # If we're do-release-upgrad-ing to bionic, then clean up the xenial-only
+    # cloud-id-shim unit
+    if [ "$VERSION_ID" = "18.04" ]; then
+        if echo "$PREVIOUS_PKG_VER" | grep -q "16.04"; then
```
Comment (basak) [inline debian/ubuntu-advantage-tools.postinst:363]: This match against 16.04 seems fragile to me, but you can leave it as-is if you prefer.

**Review by basak — APPROVED** [review]
Looks good. One minor (not blocking) inline comment.

**Pull request merged**